### PR TITLE
feat(FX-4209): setup analytics for reverse image

### PR DESCRIPTION
**Pull request by dimatretyak**
Cohesion PR: artsy/cohesion#348

This PR resolves [FX-4209]

Reverse Image Search Analytics: https://www.notion.so/artsy/reverse-image-search-analytics-d8fcb365752f4cfe9ad326f2280e8b39

### Acceptance Criteria
* A user taps the reverse image search button on a fair or show screen - `tappedReverseImageSearch`
* A user taps the flash button on the reverse image search camera screen - `tappedToggleCameraFlash`
* A user taps the library button on the reverse image search camera screen - `tappedPickImageFromLibrary`
* A user searches with a reverse image query with results (for single and multiple results) - `searchedReverseImageWithResults`
* A user searches with a reverse image query with no results - `searchedReverseImageWithNoResults`
* A user selects an artwork from reverse image search results - `selectedArtworkFromReverseImageSearch`

#### `tappedReverseImageSearch`
https://user-images.githubusercontent.com/3513494/189173757-b492910d-d2ba-434f-8eb0-fd1013964b44.MP4

#### `tappedPickImageFromLibrary`
https://user-images.githubusercontent.com/3513494/189173970-53608a5d-7b97-4702-9302-fd4066b7d438.MP4

#### `tappedToggleCameraFlash`
https://user-images.githubusercontent.com/3513494/189174900-22756c48-b21c-45fb-aa30-dc1a4fa27cf0.MP4

#### `searchedReverseImageWithResults`
https://user-images.githubusercontent.com/3513494/189175671-7051b543-bff0-40d3-b85e-cfa9b1ff7acf.MP4

#### `searchedReverseImageWithNoResults`
https://user-images.githubusercontent.com/3513494/189175932-1487db72-4e55-4c1c-923c-4f31d3ec00ee.MP4

#### `selectedArtworkFromReverseImageSearch`
https://user-images.githubusercontent.com/3513494/189176270-f0849eec-f707-4f7b-abb9-5b5cab5fa1f0.MP4

#### Acceptance Criteria

-

#### Setup Instructions

-

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [x] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [x] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- setup analytics for reverse image - dimatretyak

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[FX-4209]: https://artsyproduct.atlassian.net/browse/FX-4209?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/ReverseImage/Screens/MultipleResults/ReverseImageArtworksRail.tsx [hunk 1]
```diff
@@ -20,8 +28,17 @@ export const ReverseImageArtworksRail: React.FC<ReverseImageArtworksRailProps> =
...
 
       <SmallArtworkRail
-        artworks={artworks}
-        onPress={(artwork) => {
+        artworks={nodes}
+        onPress={(artwork, index) => {
+          tracking.trackEvent(
+            tracks.selectedArtworkFromReverseImageSearch({
+              owner,
+              position: index,
```
Comment (dimatretyak) [inline src/app/Scenes/ReverseImage/Screens/MultipleResults/ReverseImageArtworksRail.tsx:36]: **question**: should the first position start with `0` or `1`?

**Review by gkartalis — DISMISSED** [review]
Looking great! Thank you for the tests 👌

**Comment by gkartalis** [comment]
also merged -> https://github.com/artsy/eigen/pull/7324, if you rebase you should be good to go

File: src/app/Scenes/ReverseImage/Screens/MultipleResults/ReverseImageArtworksRail.tsx (hunk 1 above)
Comment (dblandin) [reply src/app/Scenes/ReverseImage/Screens/MultipleResults/ReverseImageArtworksRail.tsx:36]: For analytics, I believe they want the first position as 1. Should be able to double check by referring to the use of the tap on artwork group event.

**Review by dblandin — APPROVED** [review]
LGTM! Left a comment regarding the position value.

**Pull request merged**